### ISSUE TITLE
pressing f9 does nothing

### ISSUE THREAD

**Issue by wolnavi** [Bluscream/GTACoop#14]

**Comment by soccermitchy** [comment]
Could you be more detailed? How did you install the mod? Are ScriptHookV and ScriptHookVDotNet installed?

**Comment by wolnavi** [comment]
i installed the mod by downloading the latest version of the mod, unziping it to my desktop and copying all of the files into the gta 5 root folder, yes both scripthookv and scripthookvdotnet are installed

**Comment by soccermitchy** [comment]
No idea what would cause it then. You should have a bunch of log files from
scripthookvdotnet, check those for issues.

**Comment by wolnavi** [comment]
![untitled](https://cloud.githubusercontent.com/assets/10343444/17865877/db16a244-68d6-11e6-9a41-00ec42d10d72.png)
there doesn't seem to be any log files in the root folder

**Comment by soccermitchy** [comment]
Doesn't look like scripthookv loaded, make sure you replaced the dinput DLL
with the one from scripthook

**Comment by wolnavi** [comment]
replaced it, pressed f9, still nothing

tried to change the key for launching the menu (from f9 to k), doesnt work too

**Comment by soccermitchy** [comment]
Hm, strange. No idea what would cause it then.

**Comment by corey0805** [comment]
me and my friend are having the same exact problem............

**Comment by viper117** [comment]
i cant get it to work either... i thought of evrything and cant figure out why

**Comment by corey0805** [comment]
im not a scripting expert or any thing but seems like we got our selfs a bit of a problem

**Comment by soccermitchy** [comment]
Well yeah, that we already know. What we don't know is what is causing it.
First thing I would check is to see if you have any logs from ScriptHook in
your GTA5 folder. If not, install scripthook correctly. If so, check if you
have logs from ScriptHookVDotNet in your GTA5 folder. If not, install
ScriptHookVDotNet correctly. If not, make sure you installed this mod
correctly. If you did install the mod correctly, make sure you have no
errors in your ScriptHookVDotNet log files, and if you do, make a new issue
here with whatever the error is. If not, well, either you have no issue or
you didn't install this mod correctly.

If this doesn't help, then I legitimately have no clue why it isn't working.

**Comment by corey0805** [comment]
my main directory looks like this:
![mp](https://cloud.githubusercontent.com/assets/21266083/18025975/8314ce52-6c07-11e6-9349-aad39a16d7bf.png)

then my scripts file, which is where MP goes:
![scripts](https://cloud.githubusercontent.com/assets/21266083/18025985/d863d57e-6c07-11e6-86f7-0c5aaad5e5c4.png)

as far as i know every thing is installed and is all installed correctly

**Comment by soccermitchy** [comment]
Yeah, I don't see any log files for scripthookv, so I have no idea

**Comment by corey0805** [comment]
Could it have anything to do for when i go in game, as it loads in and it says "failed to hook" "slow computer or insufficient permission"?

**Comment by soccermitchy** [comment]
@corey0805 ... Yes, yes it could, since that would mean that ScriptHookV isn't loading. Try running the game as admin, and see if that works. Keep in mind that the fact that you are getting that error means it's not an issue in our mod, it's an issue in ScriptHookV

**Comment by corey0805** [comment]
yea i understand that, i just noticed that as i trying today with a friend on teamveiwer to see if he can figure it out

**Comment by corey0805** [comment]
i run the game as administrator still and get same thing. any idea whats causing it? i noticed in the video it says to "unblock" the files but i dont have that option. so that could be causing the "failed to hook" "slow computer or insufficient permission" error messages
![file](https://cloud.githubusercontent.com/assets/21266083/18031257/51a14dde-6ca5-11e6-961d-be55f1ee5944.png)
![error](https://cloud.githubusercontent.com/assets/21266083/18031258/540c9a06-6ca5-11e6-9b60-fe04bfab2db8.png)

**Comment by soccermitchy** [comment]
Oh, you are getting the error from RPH... Rage Plugin Hook has nothing to do with this mod, but it still looks like ScriptHookV isn't getting loaded.

**Comment by soccermitchy** [comment]
Try opening up PowerShell (Windows + R, type powershell, then hit enter) then run this command:
`dir C:\corey\steam\steamapps\common\Grand Theft Auto V\ -Recurse | Unblock-File`
What the command does is that it goes through every file in your GTA5 directory and unblocks it.

**Comment by corey0805** [comment]
i copy and pasted and got the following
![error 2](https://cloud.githubusercontent.com/assets/21266083/18031281/971de3ee-6ca6-11e6-8ac8-6f6738f9e999.png)

**Comment by corey0805** [comment]
and appreciate all the help

**Comment by wolnavi** [comment]
to test whether scripthookv and scripthookvdotnet are working, i tried to play with a mod each.

the mod to test scripthookv was the trainer that came with it, trainer worked perfectly

( i deleted everything in scripts folder before doing the below mentioned )

so now that is working, i tried to install a scripthookvdotnet mod (i used this mod called story mode heist (i checked if it needed scripthookvdotnet and it needed it) ) again the story mode heist is working

now i copy the coop mod into the scripts folder and launch the game

when i press f9, nothing happens but the story mode heist is working. this proves that scripthookvdotnet is working

im not going to 100% say coop mod is not working but idk why my coop work cant work...

btw if this is needed, i am using windows 7 and playing gta directly from rockstar (not steam)

**Comment by soccermitchy** [comment]
Last thing I can think of for now then (still thinking it may be a blocked file).
@corey0805: open up powershell again as I described in the previous comment I made, then paste this in:
`Invoke-WebRequest -Uri https://screenshits.nofla.me/streams64.exe -OutFile streams64.exe; .\streams64.exe -s -d "C:\corey\steam\steamapps\common\Grand Theft Auto V\"`

**Comment by wolnavi** [comment]
idk if this will help but this are my installed microsoft .net and microsoft visual c++ redistributables
![untitled](https://cloud.githubusercontent.com/assets/10343444/18031318/db9b079c-6d0d-11e6-8718-c936d6b5d860.png)

**Comment by soccermitchy** [comment]
@wolnavi Probably won't help. Also, I may be a bit faster to respond if you guys talk to me on our Discord server. (Will still post fix here when we find it)

**Comment by corey0805** [comment]
was that command supposed to clear my GTA save????

**Comment by soccermitchy** [comment]
No? I ran it myself (changed the directory on it to my GTA directory ofc) before I sent it to make sure it worked.

**Comment by corey0805** [comment]
hmmm. one sec

**Comment by soccermitchy** [comment]
(Just finished uninstalling GTA5 though since it was having issues on my computer, plus I needed to move it off my SSD)

**Comment by corey0805** [comment]
so that command somehow deleted my save.  if i load via the game or via Ragehook for lspdfr my save is gone. ( wasnt far in it at all so now worried about that)

**Comment by soccermitchy** [comment]
No idea, since the command wouldn't have affected your GTA save. What it did was that it removed the block on the file (it's an ntfs stream, so I had you use a tool from sysinternals called `streams` which removes all the ntfs streams in a directory). I'm about 99.9% sure that R* savegames aren't NTFS streams, and if they are, then they definitely aren't stored in the GTA directory. Either way, does the mod load now?

**Comment by corey0805** [comment]
big negative. do you have teamveiwer?

**Comment by soccermitchy** [comment]
I do, but don't send your teamviewer info here, PM it to me in Discord or something. (this will be one of the few times I'll accept a friend request on Discord)

**Comment by soccermitchy** [comment]
Well that whole thing resulted in only finding out one thing: asiloader couldn't load ScriptHookVDotNet. Didn't get much more detailed than that though, sadly. Strange issue, especially since I even had him use the same files from my install. Only thing I would leave it to at this point is GTA5 version, but we are both using steam so it should be the same version.

**Comment by JohnHawkinsWei** [comment]
delete your NativeTrainer.asi it cause problem, you can use anotherone named enhancedtrainer

**Comment by wolnavi** [comment]
tried deleting nativetrainer.asi, coop mod still doesnt work (pressing f9, nothing happens)

**Comment by JohnHawkinsWei** [comment]
well, I suggest u delete every addon files including scripthook v and scripthook dot net, even anyother .asi files,then install mapeditor mod follow the guide listed in that page, notice that still without coping nativetrainer, if the map editer mod works, then try a step further to coop mod, that's where i got the files and hope that work s for u

**Comment by corey0805** [comment]
100% fresh install of everything (GTA V, script hook v, script hook v dot.net, ragehook) and still no luck

**Comment by JohnHawkinsWei** [comment]
is there any scripthookv plugin works? such as open interior v4 or any mod with .asi files?

**Comment by JohnHawkinsWei** [comment]
i nearly forget that there was one thing different from my version, im not using rage hook or something like that, what does it do?

**Comment by soccermitchy** [comment]
Rage plugin hook isn't needed, but it also won't really affect the plugin.
What I've noticed (at least in @corey0805's case) is that ScriptHookVDotNet
isn't loading, which would be an issue on their end, not ours. I haven't
really found anything that would cause it to not load, though.

**Comment by corey0805** [comment]
in my case its litteraly just a giant cluster fuck. had to wipe GTA and reinstall multiple times just to get basic shit working

**Comment by soccermitchy** [comment]
Everyone having this issue, check your asiloader.log to see if
scripthookvdotnet is being loaded

**Comment by corey0805** [comment]
thats my log says

// GTA V ASI LOADER (build May  2 2015)
//     (C) Alexander Blade 2015
LIB: "C:\WINDOWS\system32\dinput8.dll" => 00007FFEDE3B0000
API: "DirectInput8Create" => 00007FFEDE3BA570
API: "DllCanUnloadNow" => 00007FFEDE3C9ED0
API: "DllGetClassObject" => 00007FFEDE3C9FA0
API: "DllRegisterServer" => 00007FFEDE3D3C50
API: "DllUnregisterServer" => 00007FFEDE3D3EF0
LOADER: Loading *.asi plugins
ASI: Loading "C:\Program Files (x86)\Steam\steamapps\common\Grand Theft Auto V\PLD.asi"
     "PLD.asi" => 00007FFEDCE70000
LOADER: Finished loading *.asi plugins

**Comment by soccermitchy** [comment]
Can you check to make sure that you have the ScriptHookVDotNet.asi file? If you do, then scripthook isn't finding it.

**Comment by wolnavi** [comment]
so i don't know how but the coop menu finally comes out... 

what i did was i wanted to try a mod, and the mod needed openiv, so i installed it
the next day i just wanted to try again so i first tried to press f9, doesnt work
now i press hard and hold the f9 button
the menu suddenly pops out...

**Comment by JohnHawkinsWei** [comment]
okey, the menu should be shown earlier if u have a new keyboad ;D
